ansible/roles/test/files/ptftests/advanced-reboot.py

```diff
@@ -501,24 +574,93 @@ def generate_bidirectional(self):
                 from_port = src_port
             self.packets_list.append((from_port, str(packet)))
 
+    def get_neigh_info(self):
+        for key in self.vm_dut_map.keys():
+            if self.vm_dut_map[key]['mgmt_addr'] == self.neigh_down_vm:
+                return key
+
+    def prebootOper(self):
+        # select a VM
+        vm_index = datetime.datetime.now().day % len(self.ssh_targets)
+        self.neigh_down_vm = self.ssh_targets.pop(vm_index)
```

Comment:
Can you make select_vm a function and add more comments to state reason?

Reply:
will take care of that

Reply:
This is moved to a separate file now. vm selection is a function